lib/srv/sess.go

```diff
@@ -1706,3 +1739,20 @@ func (s *session) trackSession(ctx context.Context, teleportUser string, policyS
 
 	return nil
 }
+
+// emitAuditEvent emits audit events.
+func (s *session) emitAuditEvent(ctx context.Context, event apievents.AuditEvent) error {
+	rec := s.Recorder()
+	select {
+	case <-rec.Done():
```

Comment:
Is this actually concurrency safe?

Is there a problem if two concurrent calls run, and one reads `s.Recorder()` right before the other sees `<-rec.Done()` and overwrites the recorder with a new one?